client/client.go

```diff
@@ -58,31 +60,27 @@ func (kc *KalaClient) decode(body io.Reader, target interface{}) error {
 }
 
 func (kc *KalaClient) url(parts ...string) string {
-	for i, part := range parts {
-		if part == "" {
-			continue
-		}
-		if init, last := part[:len(part)-1], part[len(part)-1]; last == '/' {
-			parts[i] = init
-		}
-	}
 	return strings.Join(append([]string{kc.apiEndpoint}, parts...), "/") + "/"
 }
 
-func (kc *KalaClient) do(method, url string, expectedStatus int, payload, target interface{}) (*http.Response, error) {
+func (kc *KalaClient) do(method, url string, expectedStatus int, payload, target interface{}) (int, error) {
```

Comment:
Make this a named return value so it's clear what the `int` being returned signifies